File: proyecto_java/iniciales/TP_bucles/ejercicio1.java
package iniciales.TP_bucles;

import java.util.Scanner;

// Escribir un programa que muestre el eco de todo
// lo que el usuario introduzca hasta que el usuario escriba "salir"

public class ejercicio1 {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        String texto = "";

        while (!texto.equals("salir")){
            System.out.println("Ingrese texto: ");
            texto = teclado.nextLine();

            if (!texto.equals("salir")){
                System.out.println(texto);
            }
        }

        teclado.close();
    }
}

File: proyecto_c++/iniciales/TP_bucles/ejercicio1.cpp
// Escribir un programa que muestre el eco de todo
// lo que el usuario introduzca hasta que el usuario escriba "salir"

#include <iostream>
#include <string>

using std::cout;
using std::endl;
using std::cin;
using std::string;
using std::getline;

int main(){
    
    string texto;

    cout << "Ingrese un texto: " << endl;
    getline(cin, texto);

    while(texto != "salir"){
        
        cout << texto << endl;
        
        cout << "Ingrese un texto: " << endl;
        getline(cin, texto);
        cout << "saliendo del programa...";
    }

    return 0;
}